components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1697,10 +1726,29 @@ public void deleteGroup(String groupId) throws NotFoundException, CharonExceptio
     @Override
     public List<Object> listGroupsWithGET(Node rootNode, int startIndex, int count, String sortBy, String sortOrder,
             String domainName, Map<String, Boolean> requiredAttributes)
+
+            throws CharonException, NotImplementedException, BadRequestException {
+        if (sortBy != null || sortOrder != null) {
+            throw new NotImplementedException("Sorting is not supported");
+        } else if (startIndex != 1 && count >= 0) {
+            throw new NotImplementedException("Pagination is not supported");
+        } else if (rootNode != null) {
+            return filterGroups(rootNode, requiredAttributes);
+        } else {
+            return listGroups(requiredAttributes);
+        }
+    }
+
+    @Override
+    public List<Object> listGroupsWithGET(Node rootNode, Integer startIndex, Integer count, String sortBy,
+            String sortOrder, String domainName, Map<String, Boolean> requiredAttributes)
             throws CharonException, NotImplementedException, BadRequestException {
+
         if (sortBy != null || sortOrder != null) {
             throw new NotImplementedException("Sorting is not supported");
-        } else if (startIndex != 1) {
+        } else if(startIndex == null){
+            throw new BadRequestException("NULL argument for startIndex");
```

Comment:
startIndex is an optional argument so isn't it can't be null? Here null means user didn't specify right?

Reply:
@sarubi +1. Better to use a debug log for dev related information, and end-user friendly message in the returned exception.

Reply:
Fixed: df77256698c74c829147f73d00e2f63979199c16